Restructuring and repairing the warehouse tutorial (#767)
cal-itp/data-infra: docs/analytics_examples/01_warehouse_tutorial.md

| -------- | -------- | -------- |
| **Dim Feeds** | Warehouse: **views.gtfs_schedule_dim_feeds** <br/> Metabase: **Gtfs Schedule Dim Feeds** | - Joining with this table is the most common way to append calitp_feed_name to fact tables <br/> - calitp_feed_name is our primary agency identifier |
````

### Important Column Types

| Column Type | Notable Columns | Description |
| -------- | -------- | -------- |
| **Agency** | Warehouse: **calitp_feed_name** <br/> Metabase: **Calitp Feed Name** | - Our primary agency identifier <br/> - In most of the examples below, this is gathered from the table: **views.gtfs_schedule_dim_feeds** <br/> - Metabase: **Gtfs Schedule Dim Feeds** |
| **Time** | | |
| **Geography** | | |

### Python Libraries to Import

```{code-cell}
from calitp.tables import tbl
from calitp import query_sql
from siuba import *
import pandas as pd
import calitp.magics
pd.set_option("display.max_rows", 20)
```

```{code-cell}
:tags: [remove-cell]
from myst_nb import glue
```

## Query Examples

(routes-agency-time)=
### 1. Number of Routes for a Given Agency Over Time

```{code-cell}
:tags: [remove-cell]
df_routes = query_sql("""
SELECT
    calitp_feed_name,
    date,

>     count(*) AS count_feeds

Reviewer: This is counting the number of routes, right? Seems like this should be:

```suggestion
    count(*) AS count_routes
```

LIMIT 10
```

```{glue:figure} df_feed_expires_output
```
````

````{tabbed} siuba
**Tables Used**
| Name | Use |
| -------- | -------- |
| **views.gtfs_schedule_fact_daily_feeds** | Primary Fact Table |
| **views.gtfs_schedule_dim_feeds** | Secondary Dimensional Table |

**Important Columns**
| Type | Column | Use |
| -------- | -------- | -------- |
| *Time* | **date** | *select*, *filter* by |
| *Time* | **feed_end_date** | The date that the particular feed expires, to *select* |
| *Measure* | **days_until_feed_end_date** | The value we are trying to find for the particular date, to *select* |
| *Agency* | **calitp_feed_name** | *left_join* with table **views.gtfs_schedule_dim_feeds** on variable **feed_key** for **calitp_feed_name**, to *select* and *filter* by |

```python
(
    tbl.views.gtfs_schedule_fact_daily_feeds()
    >> left_join(_, tbl.views.gtfs_schedule_dim_feeds(), "feed_key")
    >> select(_.calitp_feed_name, _.date, _.days_until_feed_end_date, _.feed_end_date)
    >> filter(_.date == "2021-09-01", _.calitp_feed_name == "Unitrans (0)")
)
```
```{glue:figure} siuba_feed_expires_output
```
````

(max-number-stops)=

> ### 5. Max Number of Stops a Trip Can Have, Per Agency

Reviewer: Query description suggestion:

```suggestion
### 5. Find the Trip With the Most Number of Stops, Per Agency
```
